src/x86_64.rs

```diff
@@ -0,0 +1,331 @@
+///
+/// pub struct kvm_segment {
+///    pub base: __u64,
+///    pub limit: __u32,
+///    pub selector: __u16,
+///    pub type_: __u8,
+///    pub present: __u8,
+///    pub dpl: __u8,
+///    pub db: __u8,
+///    pub s: __u8,
+///    pub l: __u8,
+///    pub g: __u8,
+///    pub avl: __u8,
+///    pub unusable: __u8,
+///    pub padding: __u8,
+/// }
+pub use kvm_bindings::kvm_segment as SegmentRegister;
+
+/// Descriptor Table
+///
+/// pub struct kvm_dtable {
+///     pub base: __u64,
+///     pub limit: __u16,
+///     pub padding: [__u16; 3usize],
+/// }
+pub use kvm_bindings::kvm_dtable as DescriptorTable;
+
+/// Floating Pointe Unit State
+///
+/// pub struct kvm_fpu {
+///    pub fpr: [[__u8; 16usize]; 8usize],
+///    pub fcw: __u16,
+///    pub fsw: __u16,
+///    pub ftwx: __u8,
+///    pub pad1: __u8,
+///    pub last_opcode: __u16,
+///    pub last_ip: __u64,
+///    pub last_dp: __u64,
+///    pub xmm: [[__u8; 16usize]; 16usize],
+///    pub mxcsr: __u32,
+///    pub pad2: __u32,
+/// }
+pub use kvm_bindings::kvm_fpu as FpuState;
+
+/// Entry describing a CPUID feature/leaf. Features can be set as responses to
+/// the CPUID instruction.
+/// pub struct kvm_cpuid_entry2 {
+///    pub function: __u32,
+///    pub index: __u32,
+///    pub flags: __u32,
+///    pub eax: __u32,
+///    pub ebx: __u32,
+///    pub ecx: __u32,
+///    pub edx: __u32,
+///    pub padding: [__u32; 3usize],
+/// }
+use kvm_bindings::kvm_cpuid_entry2 as CpuIdEntry2;
+
+/// Array of CpuId2 entries, each of which describing a feature/leaf to be set
+/// pub struct kvm_cpuid2 {
+///    pub nent: __u32,
+///    pub padding: __u32,
+///    pub entries: __IncompleteArrayField<kvm_cpuid_entry2>,
+/// }
+use kvm_bindings::kvm_cpuid2 as CpuId2;
+
+/// Unix definition of the LAPIC state, the set of memory mapped registers that
+/// describe the Local APIC. Unix-based VMMs only require 1KB of memory to
+/// describe the LAPIC state.
+///
+/// pub struct kvm_lapic_state {
+///    pub regs: [::std::os::raw::c_char; 1024usize],
+/// }
+#[cfg(unix)]
+pub use kvm_bindings::kvm_lapic_state as LapicState;
+
+/// Windows definition of the LAPIC state, the set of memory mapped registers
+/// that describe the Local APIC. Windows-based VMMs require 4KB of memory to
+/// describe the LAPIC state, or the Windows APIs will fail, even though the
+/// architecture-specified space requirement is only 1KB.
+#[cfg(windows)]
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct LapicState {
+    pub regs: [::std::os::raw::c_char; 4096usize],
+}
+
+#[cfg(windows)]
+impl Default for LapicState {
+    fn default() -> Self {
+        unsafe { ::std::mem::zeroed() }
+    }
+}
+
+#[cfg(windows)]
+impl ::std::fmt::Debug for LapicState {
+    fn fmt(&self, fmt: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+        self.regs[..].fmt(fmt)
+    }
+}
+
+#[test]
+fn vcpu_test_layout_lapic_state() {
+    assert_eq!(
+        ::std::mem::size_of::<LapicState>(),
+        1024usize,
```

Comment:
This will fail on Linux. The test code should be label with cfg(windows).

Reply:
Fixed